api/src/controllers/document-controller.js

```diff
@@ -254,13 +263,153 @@ async function createDocument(fileName, addedBy, url = null) {
   document.addedBy = addedBy;
   document.url = url;
   document.key = `${document._id}/${fileName}`;
-  document.read = ['public', 'sysadmin'];
+  document.read = ['sysadmin'];
   document.write = ['sysadmin'];
+
+  if (!isAnonymous) {
+    document.read.push('public');
+  }
+
```

Comment:
I don't think whether the individual/company is anonymous or not should determine the document publishing rule.  It should be separate from the anonymous logic, and controlled via a different flag.  Maybe just the terminology is wrong, but the roles to be added should probably be passed into the function, as opposed to a simple binary flag.